File: code/547.go
/*
 * @Author: lmio
 * @Date: 2023-02-22 22:12:24
 * @LastEditTime: 2023-02-22 22:22:40
 * @FilePath: /leetcode/code/547.go
 * @Description:547.省份数量
 */
package code

func FindCircleNum(isConnected [][]int) int {
	n, m := len(isConnected), len(isConnected[0])
	res := 0
	var dfs func (x, y int)
	dfs = func (x, y int)  {
		if isConnected[x][y] != 1 {
			return
		}
		isConnected[x][y] = 0
		for i := 0; i < m; i++ {
			dfs(y, i)
		}
	}
	for i := 0; i < n; i++ {
		for j := 0; j < m; j++ {
			if isConnected[i][j] == 1 {
				res++
				dfs(i, j)
			}
		}
	}
	return res
}

File: code/547.py
'''
Author: lmio 2091319361@qq.com
Date: 2023-10-09 23:18:13
LastEditors: lmio 2091319361@qq.com
Description: 547. 省份数量
'''

from typing import List


class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        n = len(isConnected)
        res = 0
        def dfs(x, y):
            if isConnected[x][y] != 1:
                return
            isConnected[x][y] = 0
            for i in range(n):
                dfs(y, i)
        for i in range(n):
            for j in range(n):
                if isConnected[i][j] == 1:
                    res += 1
                    dfs(i, j)
        return res